Physx collider component (#787)
o3de/o3de.org: content/docs/user-guide/gems/reference/physics/nvidia/physx.md

toc: true
---

The PhysX Gem provides physics simulation with NVIDIA PhysX. With PhysX, you can create static, dynamic, and kinematic rigid body simulation, force regions, ragdolls, and dynamic PhysX joints. The PhysX Gem also provides functionality for creating trigger volumes, and performing overlap tests, shapecasts, and raycasts.

## Installation

To use PhysX, enable the PhysX Gem in **Project Manager**. The PhysX Gem requires the NVIDIA PhysX SDK package, which is automatically downloaded as part of the configuration and build process. 

> ## Gem Contents

Reviewer: ```suggestion
## Gem contents
```

In addtion to providing phyiscs simulation with PhysX, the PhysX Gem provides several components, a configutation tool, scene processing features, and the phyiscs material asset template, so that you can create and tune physics simulations for your projects.

### PhysX components

> The components included in the PhysX Gem are used to create static, dynamic, and kinematic colliders, triggers, force reqions, dynamic joints, and physical actor behaviors.

Reviewer: ```suggestion
The components included in the PhysX Gem are used to create static, dynamic, and kinematic colliders, triggers, force regions, dynamic joints, and physical actor behaviors.
```

| [PhysX Character Controller](/docs/user-guide/components/reference/physx/character-controller/) | Implements basic character interactions with the physical world. For example, the PhysX Character Controller can define interactions with slopes and steps, set minimum movement distance and maximum speed, and prevent characters from walking through walls or passing through terrain. |
| [PhysX Character Gameplay](/docs/user-guide/components/reference/physx/character-gameplay/) | Provides example implementations for character controller behaviors which are likely to require game-specific tweaking, such as detecting whether the character is on the ground, interacting with gravity, and behavior for interacting with kinematic bodies and other character controllers. |
| [PhysX Ragdoll](/docs/user-guide/components/reference/physx/ragdoll/) | Enables physical simulation on rigged actors of behaviors such as hit reactions. The ragdoll is represented by a hierarchy of rigid bodies connected by joints. |

### PhysX Configuration tool

> The PhysX Configuration tool is accessed through the **Tools** menu. You can set various global properties of PhysX simulation including the default physics material library, simulation time steps, buffer sizes for scene queries such as raycasts, default gravity, and much more. You can also define collision layers and groups, as well as set PhysX Visual Debugger settings in the PhysX configuration tool.

Reviewer 2: Switch to active voice. 
```suggestion
You can access the PhysX Configuration tool through the **Tools** menu. You can set various global properties of PhysX simulation including the default physics material library, simulation time steps, buffer sizes for scene queries such as raycasts, default gravity, and much more. You can also define collision layers and groups, as well as set PhysX Visual Debugger settings in the PhysX configuration tool.
```

### PhysX scene processing

> The PhysX Gem provides PhysX scene processing features so you can export meshes you create, automatically generate covex hulls, or automatically fit shapes to decomposed complex meshes, to define colliders for entities in PhysX simulations.

Reviewer: ```suggestion
The PhysX Gem provides PhysX scene processing features so you can export meshes you create, automatically generate convex hulls, or automatically fit shapes to decomposed complex meshes, to define colliders for entities in PhysX simulations.
```
